lib/sycamore/sycamore/data/document.py

```diff
@@ -301,6 +303,66 @@ def split_data_metadata(all: list[Document]) -> tuple[list[Document], list[Metad
     )
 
 
+############### EXPERIMENTAL
+class HierarchicalDocument(Document):
+    def __init__(self, document=None, **kwargs):
+        super().__init__(document)
+        if "doc_id" not in self.data:
+            self.doc_id = str(uuid.uuid4())
+        if "children" not in self.data:
+            self.children = []
+        if "elements" in self.data:
+            for element in self.data["elements"]:
+                self.children.append(HierarchicalDocument(Document(element.data)))
+
+        del self.data["elements"]
+
+    @property
+    def children(self) -> list["HierarchicalDocument"]:
+        """TODO"""
+        return self.data["children"]
+
+    @children.setter
+    def children(self, children: list["HierarchicalDocument"]):
+        """TODO"""
+        self.data["children"] = children
+
+    @children.deleter
+    def children(self) -> None:
+        """TODO"""
+        self.data["children"] = []
+
+    @property
+    def elements(self) -> list[Element]:
+        raise ValueError("MetadataDocument does not have elements")
+
+    @elements.setter
+    def elements(self, elements: list[Element]):
+        raise ValueError("MetadataDocument does not have elements")
+
+    def __str__(self) -> str:
```

Comment:
I tried putting it in the base class, but the base class does not have the children property that I use in HierarchicalDocument. My hope was that as we transition over to a hierarchical data model, this would not be a seperate class anymore so the bloat wouldn't be as big.